iceoryx2-bb/posix/src/mutex.rs

```diff
@@ -177,31 +178,40 @@ impl From<MutexCreationError> for MutexError {
 /// }
 /// ```
 #[derive(Debug)]
-pub struct MutexGuard<'mutex, 'handle, T: Debug> {
-    mutex: &'mutex Mutex<'handle, T>,
+pub struct MutexGuard<'handle, T: Debug> {
+    handle: &'handle MutexHandle<T>,
 }
 
-unsafe impl<T: Send + Debug> Send for MutexGuard<'_, '_, T> {}
-unsafe impl<T: Send + Sync + Debug> Sync for MutexGuard<'_, '_, T> {}
-
-impl<T: Debug> Deref for MutexGuard<'_, '_, T> {
+impl<T: Debug> Deref for MutexGuard<'_, T> {
     type Target = T;
 
     fn deref(&self) -> &Self::Target {
-        unsafe { (*self.mutex.handle.value.get()).as_ref().unwrap() }
+        unsafe { (*self.handle.value.get()).as_ref().unwrap() }
     }
 }
 
-impl<T: Debug> DerefMut for MutexGuard<'_, '_, T> {
+impl<T: Debug> DerefMut for MutexGuard<'_, T> {
     fn deref_mut(&mut self) -> &mut Self::Target {
-        unsafe { (*self.mutex.handle.value.get()).as_mut().unwrap() }
+        unsafe { (*self.handle.value.get()).as_mut().unwrap() }
     }
 }
 
-impl<T: Debug> Drop for MutexGuard<'_, '_, T> {
+impl<T: Debug> MutexGuard<'_, T> {
+    pub(crate) fn release(&self) -> Result<(), MutexUnlockError> {
```

Comment:
This function was moved from `Mutex` to `MutexGuard`. Required since the `MutexGuard` now holds the handle and no longer a reference to the `Mutex` and calls `release` on `Drop`.